sympy/utilities/_compilation/tests/test_compilation.py

```diff
@@ -60,3 +64,23 @@ def test_compile_link_import_strings():
     finally:
         if info and info['build_dir']:
             shutil.rmtree(info['build_dir'])
+
+
+def test_compile_sources():
+    build_dir = tempfile.mkdtemp("sympy_test_compilation")
+    def _cleanup():
+        shutil.rmtree(build_dir)
+    atexit.register(_cleanup)
+    _handle, file_path = tempfile.mkstemp('.c', dir=build_dir)
+    with open(file_path, 'wt') as ofh:
+        ofh.write("""
+        int foo(int bar) {
+            return 2*bar;
+        }
+        """)
+    obj, = compile_sources([file_path], cwd=build_dir)
+    obj_path = get_abspath(obj, cwd=build_dir)
+    assert os.path.exists(obj_path)
+    if os.name == 'posix':
+        nm_out = subprocess.check_output(["nm", obj_path])
+        assert 'foo' in nm_out.decode('utf-8')
```

Comment:
Rather than checking `os.name` it would be better to catch an exception to handle the case that `nm` is not installed e.g. by checking if `nm --help` works first.

Reply:
agreed.